src/TypeScript/TsConfig/TsConfigProvider.cs

```diff
@@ -77,25 +77,33 @@ public async Task<string> GetConfigForFile(string sourceFilePath, CancellationTo
         {
             foreach (var tsConfigFilePath in candidateTsConfigs)
             {
-                if (visited.Contains(tsConfigFilePath))
+                var tsConfigToCheck = tsConfigFilePath;
+                var isDirectory = !Path.HasExtension(tsConfigToCheck);
+
+                if (isDirectory)
+                {
+                    tsConfigToCheck = Path.GetFullPath(Path.Combine(tsConfigToCheck, "tsconfig.json"));
```

Comment:
The documentation was unclear about whether the tsconfig has to be under the specified directory or if it can be in a sub folder -- but VS fails if it's not directly under, so that's what I went with as well.